src/ComputeSharp.D2D1/Shaders/Interop/D2D1ResourceTextureManager.cs

```diff
@@ -0,0 +1,487 @@
+/// can be used to initialize the resource texture held by the manager. The manager can also be assigned to an effect at any time,
+/// using the available property indices from <see cref="D2D1PixelShaderEffectProperty"/>. To update texture data after its initial
+/// creation, <see cref="Update(void*, ReadOnlySpan{uint}, ReadOnlySpan{uint}, ReadOnlySpan{uint}, ReadOnlySpan{byte})"/> can be used.
+/// </para>
+/// <para>
+/// An RCW (runtime callable wrapper, see <see href="https://docs.microsoft.com/dotnet/standard/native-interop/runtime-callable-wrapper"/>),
+/// is also available for all of these APIs, implemented by the same <see cref="D2D1PixelShaderEffect"/>. In this case, the constructor can
+/// be used to create and initialize an instance (equivalent to calling <see cref="Create"/> and <see cref="Initialize"/>, and then
+/// <see cref="Update(ReadOnlySpan{uint}, ReadOnlySpan{uint}, ReadOnlySpan{uint}, ReadOnlySpan{byte})"/> is available for texture data updates.
+/// The instance implements <see cref="ICustomQueryInterface"/>, which can be used in advanced scenarios to retrieve the underlying COM object.
+/// </para>
+/// <para>
+/// The <c>ID2D1ResourceTextureManager</c> contract allows callers to create a resource texture at any time even before the effect
+/// has been initialized and the manager assigned to it. In that case, the manager will buffer the data internally, and will defer
+/// the creation of the actual resource texture until it's possible for it to do so, at which point it will also free the buffer.
+/// </para>
+/// <para>
+/// If implementing a custom resource manager was needed, in order for external implementations to be accepted by the effect objects
+/// created by <see cref="D2D1PixelShaderEffect"/>, the <c>ID2D1ResourceTextureManagerInternal</c> interface also needs to be implemented:
+/// <code>
+/// [uuid(5CBB1024-8EA1-4689-81BF-8AD190B5EF5D)]
+/// interface ID2D1ResourceTextureManager : IUnknown
+/// {
+///     HRESULT Initialize(
+///         [in]           ID2D1EffectContext *effectContext,
+///         [in, optional] const UINT32       *dimensions);
+/// 
+///     HRESULT GetResourceTexture([out] ID2D1ResourceTexture **resourceTexture);
+/// };
+/// </code>
+/// </para>
+/// <para>
+/// Note that due to the fact <c>ID2D1EffectContext</c> and <c>ID2D1ResourceTexture</c> are not thread safe, they are only
+/// safe to use without additional synchronization when <c>Initialize</c> and <c>GetResourceTexture</c> are called.
+/// This happens from the internal <c>ID2D1EffectImpl</c> object, which is invoked by D2D, which will handle taking the
+/// necessary lock before invoking these APIs. Due to the fact the context has to be stored internally, custom implementations
+/// have to make sure to retrieve the right <c>ID2D1Multithread</c> instance from the input context and use that to synchronize
```

Comment:
proofreading nit: ... ~~input~~ effect context ...

Reply:
Fixed in 28ed19c349592ecaafaafd00328df254e7556db9.